### DISCUSSION TITLE
Custom properties in UserInfo

### DISCUSSION THREAD

**Discussion by mklimko-perpetio** [serverpod/serverpod#473] [Q&A] [answered]
Is there any way to define custom fields in serverpod_user_info table (so I can get them by `sessionManager.signedInUser?.newFieldName`)? manually or with generation?
For example I would like to add new field for user "favouriteColor" or "age". How I can do this? Should I create new table named "db_users" and store there all my custom fields or there is some better solution?

**Comment by vlidholt** [comment, marked as answer]
Yes, the recommended way is to use a second table for that.

**Comment by lodgmar** [comment]
It would be great if UserInfo had a simple JSON extra field (`Map<String, dynamic>`). Anyone needing complex queries should use a second table, but for the majority of cases, a JSON field is enough.

**Comment by ROCKWAR** [comment]
I'm using the authentication system that ServerPod provides out of the box, and it works with UuidValue IDs. However, I see that the UserInfo ID is an integer (which I don't use). I also see that it has a table called serverpod_auth_core_profile, which contains some user details like their image, email, etc. But as mentioned, to add more details to this table, we need to create another table and add the fields we need. Let's say I've created my user_details table, and that it's related to serverpod_auth_core_user. My question is, how do I detect the creation of a new user so I can create the new record in user_details? Perhaps the user creation method triggers an event on a channel I can subscribe to? Or what's the recommended approach?

**Reply by marcelomendoncasoares** [reply]
Answered at https://github.com/serverpod/serverpod/discussions/4679, but just pasting a reference here for future readers.
